protocol/src/fixed_codec/primitive.rs

```diff
@@ -65,6 +78,32 @@ impl FixedCodec for u64 {
     }
 }
 
+impl FixedCodec for u128 {
+    fn encode_fixed(&self) -> ProtocolResult<Bytes> {
+        let mut buf = [0u8; mem::size_of::<u128>()];
+        LittleEndian::write_u128(&mut buf, *self);
+
+        Ok(BytesMut::from(buf.as_ref()).freeze())
+    }
+
+    fn decode_fixed(bytes: Bytes) -> ProtocolResult<Self> {
+        Ok(LittleEndian::read_u128(bytes.as_ref()))
+    }
+}
+
+impl FixedCodec for usize {
```

Comment:
The "usize"  length differently on 32-bit and 64-bit systems, so it is not recommended.

Delete this code.

Reply:
I have already fixed the length of `usize` as 64 bytes.